src/rwlock.rs

```diff
@@ -595,12 +774,14 @@ impl<'a, T: ?Sized> RwLockUpgradableReadGuard<'a, T> {
 }
 
 impl<T: fmt::Debug + ?Sized> fmt::Debug for RwLockUpgradableReadGuard<'_, T> {
+    #[inline]
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         fmt::Debug::fmt(&**self, f)
     }
 }
 
 impl<T: fmt::Display + ?Sized> fmt::Display for RwLockUpgradableReadGuard<'_, T> {
+    #[inline]
```

Comment:
Is there any reason why you've inlined several methods here?

Reply:
Consistency with the equivalent trivial methods on the `Arc` guards. I can remove them if you like